dracut-module-setup.sh

```diff
@@ -327,11 +327,30 @@ _install_nmconnection() {
     inst "$_src" "$_dst"
 }
 
+kdump_install_ovs_nmconnection() {
+    local _uuid _uuid _cloned_nm_path _netif
+
+    [[ -n $ovs_phy_if ]] || return
+    _netif=$ovs_phy_if
+
+    while IFS=: read -r _nm_conn_path _; do
+        [[ $(nmcli --get-values connection.interface-name connection show "$_nm_conn_path") == $_netif ]] || continue
+        if _cloned_nm_path=$(clone_and_modify_nmconnection "$_netif" "$_nm_conn_path"); then
+            _install_nmconnection "$_cloned_nm_path"
+        else
+            derror "Failed to install the .nmconnection for $_netif"
+            exit 1
+        fi
+        return 0
+    done <<<"$(nmcli --get-values filename,ACTIVE connection show | grep "no$")"
+}
+
 kdump_install_nmconnections() {
     local _netif _nm_conn_path _cloned_nm_path
 
     while IFS=: read -r _netif _nm_conn_path; do
         [[ -v "unique_netifs[$_netif]" ]] || continue
+        [[ $ovs_phy_if == $_netif ]] && continue
```

Comment:
Do you really need a separate kdump_install_ovs_nmconnection? The way I see it kdump_install_nmconnection and kdump_install_ovs_nmconnection are very similar. Wouldn't it be enough  to have

`is_ovs_bridge "$_netif" && _netif=$(ovs_find_phy_if "$_netif"`

here? That would require a few more calls to is_ovs_bridge but would remove the need to have the extra function and the global variable ovs_phy_if.

Reply:
The connection profile we try to bring up in kdump initrd is actually inactive in 1st kernel, so I still think we need an extra function.

Let's me give an concrete example. For the following case, currently the kdump dracut module finds br-ex is used for network communication with remote fs,
```
sh-5.1# nmcli c show         
NAME                UUID                                  TYPE           DEVICE 
ovs-if-br-ex        e9b9247b-f3e3-4f8b-9890-969aff471eff  ovs-interface  br-ex  
br-ex               118c4996-d3b3-4350-b4b2-b16fd0e5a6eb  ovs-bridge     br-ex  
ovs-if-phys0        569596f1-69f7-4e6c-8a88-21614291ab38  ethernet       ens5   
ovs-port-br-ex      a3290363-e7f8-4d4b-ab13-c5cdae8dc121  ovs-port       br-ex  
ovs-port-phys0      141f408a-27fc-4638-a188-276ad9d0f043  ovs-port       ens5   
Wired connection 1  778a339a-4809-32c3-9abc-cc7c82ebe094  ethernet       --     
``` 
But we actually want to bring up the last inactive profile "Wired connection 1" instead.

Reply:
Hi Coiby,
you are right. My suggestion didn't take into account that we also need to get the .nmconnection for the inactive connection, which is exactly what kdump_install_ovs_nmconnection does. So we need to keep it...

Nevertheless could you move the call of kdump_install_ovs_nmconnection into kdump_install_nmconnection rather than calling it in kdump_install_net? In my opinion that makes it clearer that this is a special case for this function. Plus it should allow to get rid off the ovs_phy_if global variable.

In particular I mean this change (haven't tested it)

```
diff --git a/dracut-module-setup.sh b/dracut-module-setup.sh
index 9374548..a568489 100755
--- a/dracut-module-setup.sh
+++ b/dracut-module-setup.sh
@@ -328,10 +328,9 @@ _install_nmconnection() {
 }
 
 kdump_install_ovs_nmconnection() {
-    local _uuid _uuid _cloned_nm_path _netif
+    local _cloned_nm_path _netif
 
-    [[ -n $ovs_phy_if ]] || return
-    _netif=$ovs_phy_if
+    _netif=$(ovs_find_phy_if "$1")
 
     while IFS=: read -r _nm_conn_path _; do
         [[ $(nmcli --get-values connection.interface-name connection show "$_nm_conn_path") == $_netif ]] || continue
@@ -350,7 +349,12 @@ kdump_install_nmconnections() {
 
     while IFS=: read -r _netif _nm_conn_path; do
         [[ -v "unique_netifs[$_netif]" ]] || continue
-        [[ $ovs_phy_if == $_netif ]] && continue
+
+       if is_ovs_bridge "$_netif"; then
+               kdump_install_ovs_nmconnection "$_netif"
+               continue
+       fi
+
         if _cloned_nm_path=$(clone_and_modify_nmconnection "$_netif" "$_nm_conn_path"); then
             _install_nmconnection "$_cloned_nm_path"
         else
@@ -569,11 +573,6 @@ kdump_collect_netif_usage() {
         kdump_setup_vlan "$_netdev"
     fi
 
-    if is_ovs_bridge "$_netdev"; then
-        ovs_phy_if=$(ovs_find_phy_if "$_netdev")
-        _netdev=$ovs_phy_if
-    fi
-
     _save_kdump_netifs "$_netdev"
 
     if [[ ! -f ${initdir}/etc/cmdline.d/50neednet.conf ]]; then
@@ -627,7 +626,6 @@ kdump_install_net() {
     _netifs=$(_get_kdump_netifs)
     if [[ -n "$_netifs" ]]; then
         kdump_install_nmconnections
-        kdump_install_ovs_nmconnection
         apply_nm_initrd_generator_timeouts
         kdump_setup_znet "$_netifs"
         kdump_install_nm_netif_allowlist "$_netifs"
@@ -1049,7 +1047,7 @@ remove_cpu_online_rule() {
 
 install() {
     declare -A unique_netifs ipv4_usage ipv6_usage
-    local arch ovs_phy_if
+    local arch
 
     kdump_module_init
     kdump_install_conf
```

Reply:
Thanks for the suggestion! As I switch to another approach (as explained to [the second to last comment](https://github.com/rhkdump/kdump-utils/pull/2#issuecomment-2090628732)), the suggestion no longer applies. The new approach still needs a global variable but the code is more modular and easier to review.